src/gas/uniswap/UniswapGas.s.sol

```diff
@@ -0,0 +1,119 @@
+    AztecTypes.AztecAsset internal wethAsset;
+    AztecTypes.AztecAsset internal daiAsset;
+
+    UniswapBridge.SplitPath internal emptySplitPath;
+
+    function setUp() public override (BaseDeployment) {
+        super.setUp();
+
+        address defiProxy = IRead(ROLLUP_PROCESSOR).defiBridgeProxy();
+        vm.label(defiProxy, "DefiProxy");
+
+        vm.broadcast();
+        gasBase = new GasBase(defiProxy);
+
+        address temp = ROLLUP_PROCESSOR;
+        ROLLUP_PROCESSOR = address(gasBase);
+        bridge = UniswapBridge(payable(deploy()));
+        ROLLUP_PROCESSOR = temp;
+
+        ethAsset = AztecTypes.AztecAsset({id: 0, erc20Address: address(0), assetType: AztecTypes.AztecAssetType.ETH});
+        wethAsset =
+            AztecTypes.AztecAsset({id: 2, erc20Address: address(WETH), assetType: AztecTypes.AztecAssetType.ERC20});
+        daiAsset = AztecTypes.AztecAsset({
+            id: 1,
+            erc20Address: 0x6B175474E89094C44Da98b954EedeAC495271d0F,
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+
+        // List vaults and fund subsidy
+        vm.startBroadcast();
+        bridge.registerSubsidyCriteria(ethAsset.erc20Address, daiAsset.erc20Address);
+        bridge.registerSubsidyCriteria(daiAsset.erc20Address, ethAsset.erc20Address);
+        SUBSIDY.subsidize{value: 1e17}(
+            address(bridge), bridge.computeCriteria(ethAsset, emptyAsset, daiAsset, emptyAsset, 0), 500
+        );
+        SUBSIDY.subsidize{value: 1e17}(
+            address(bridge), bridge.computeCriteria(daiAsset, emptyAsset, ethAsset, emptyAsset, 0), 500
+        );
+        SUBSIDY.registerBeneficiary(BENEFICIARY);
+        vm.stopBroadcast();
+
+        // Warp time to increase subsidy
+        vm.warp(block.timestamp + 10 days);
+    }
+
+    function measure1SplitPathSwap() public {
+        uint64 path = bridge.encodePath(
+            1 ether, 1e20, WETH, UniswapBridge.SplitPath(100, 500, USDC, 100, address(0), 100), emptySplitPath
+        );
+
+        vm.broadcast();
+        address(gasBase).call{value: 2 ether}("");
+        emit log_named_uint("ETH balance of gasBase", address(gasBase).balance);
+
+        {
+            vm.broadcast();
+            gasBase.convert(
+                address(bridge), ethAsset, emptyAsset, daiAsset, emptyAsset, 1 ether, 0, path, BENEFICIARY, 300000
+            );
+        }
+
+        uint256 claimableSubsidyAfterDeposit = SUBSIDY.claimableAmount(BENEFICIARY);
+        assertGt(claimableSubsidyAfterDeposit, 0, "Subsidy was not claimed during deposit");
+        emit log_named_uint("Claimable subsidy after deposit", claimableSubsidyAfterDeposit);
+    }
+
+    function measure2SplitPathsSwap() public {
+        uint64 path = bridge.encodePath(
+            1 ether,
+            1e20,
+            WETH,
+            UniswapBridge.SplitPath(50, 500, USDC, 100, address(0), 100),
+            UniswapBridge.SplitPath(50, 500, USDC, 100, address(0), 100)
```

Comment:
When both splits are going through the same path it will not give a fitting gas usage as these contracts will be hot and storage updates to touched storage will be a lot cheaper. I suggest making the second path go to USDT or so in the middle.

Reply:
Good point will address.

Reply:
Addressed in https://github.com/AztecProtocol/aztec-connect-bridges/pull/287/commits/c7ef1bfbf76d0324e0fc4cd6ae80382d8f82370c . The gas usage jumped to 660k when I chose the paths to be as complex as possible (2 different split paths with 2 middle pools -  3 swaps).